_includes/manuals/2.4/1.2_release_notes.md

```diff
@@ -254,6 +254,11 @@ See the upgrade warnings for more information.
 
 Foreman 2.1 started to default to Puma while still allowing to fall back to Passenger. This support was kept in for users who ran into problems. Now that it has been the default for a few releases, this release deprecates Passenger support which [will be removed in the future](https://community.theforeman.org/t/drop-passenger-support/22787). Users are strongly encouraged to switch to Puma or reach out if there's something holding them back.
 
+#### Puppet ENC
+
+All of the Puppet ENC funcionality is deprecated and [will be moved to plugin in the future](https://community.theforeman.org/t/puppet-plugin-release-and-its-future/22335).
```

Comment:
I have said it again and again, but I still think this is wrong. AFAIK the ENC is still there but you can no longer assign classes. That means you still have a basic ENC that provides the Foreman information and host parameters. If you have a system where you use those (like the hostgroup) as classifiers in hiera, I think you don't need the new plugin.

If my understanding is correct, the phrase "All of the Puppet ENC functionality" is misleading.

There's also a typo in functionality.

Reply:
I'm confused. The puppet environment assignment is extracted, puppet classes and smart class parameters are extracted. The thing that remains is the general host parametrization that is e.g. used in provisioning, the other Foreman related information such as networking. However that I think can't be considered ENC (external node classifier). If I understand your statement correctly, hiera can load additional information from external inventories, such as Foreman. But the ENC in that case is hiera. I think you can't use vanilla Foreman as ENC without the foreman_puppet plugin.

Reply:
>  If I understand your statement correctly, hiera can load additional information from external inventories, such as Foreman. But the ENC in that case is hiera. I think you can't use vanilla Foreman as ENC without the foreman_puppet plugin.

Hiera can use variables. I haven't tried this, but I think it should be able to use all variables provided by the ENC. Hostgroup being the most interesting one.

> EDIT: after re-reading your comment several times, perhaps the misunderstanding is about `Puppet ENC` vs `generic ENC` concept. Do you mean general ENC functionality? This sentence mentions the very specific Puppet ENC output format which includes the puppet envrionment, puppet classes and parameters keys.

Indeed. Users might conclude that the /enc endpoint is going away but that isn't. I think in the sentence you just gave (what it includes) is already more useful to users. Let's not be too brief in this. Moving core functionality to a plugin is a big step and users will be much more comfortable with it if they are provided with more information.

So please add another paragraph that explains that the ENC endpoint is staying but what's going away.

Reply:
Ok, so we should say something like "Foreman can still provide information necessary to classify hosts, the API endpoint remains. It will no longer include the environment, smart class parameters and classes, therefore it can't serve as a Puppet ENC. If you want to keep this functionality, you need to install the newly created puppet_foreman plugin."

Reply:
I think that would help a lot.

Reply:
Endpoint is staying, but all the puppet ENC related information is going away. I'm unclear on what the users are most interested in, so I can draft it, but I'm worried to confuse more than inform. As all the relevat information is stated in the blog post and can be found for interested users. But I'll try to sum up.

Reply:
I believe I've included the info summarized in Marek's comment, if not, please let me know :)